services/core-web/src/components/modalContent/GeneratePermitNumberModal.js

```diff
@@ -0,0 +1,68 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Alert } from "antd";
+import { connect } from "react-redux";
+import { getFormSyncErrors, hasSubmitFailed } from "redux-form";
+import * as FORM from "@/constants/forms";
+import GeneratePermitNumberForm from "@/components/Forms/permits/GeneratePermitNumberForm";
+
+const propTypes = {
+  closeModal: PropTypes.func.isRequired,
+  documentType: PropTypes.objectOf(PropTypes.any).isRequired,
+  onSubmit: PropTypes.func.isRequired,
+  formSyncErrors: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.string)).isRequired,
+  signature: PropTypes.string.isRequired,
+  submitFailed: PropTypes.bool.isRequired,
+  title: PropTypes.string,
+};
+
+const defaultProps = {
+  title: "Generate Document",
+};
+
+export const GeneratePermitNumberModal = (props) => {
+  const errorsLength = Object.keys(props.formSyncErrors).length;
+  const showErrors = props.submitFailed && errorsLength > 0;
+  return (
+    <div>
+      {!props.signature && (
+        <>
+          <Alert
+            message="Signature needed to generate."
+            description="The signature for the Issuing Inspector has not been provided."
+            type="error"
+            showIcon
+          />
+          <br />
+        </>
+      )}
+      <GeneratePermitNumberForm
+        {...props}
+        showActions={false}
+        onSubmit={props.onSubmit}
+        disabled={!props.signature}
+      />
+      {showErrors && (
+        <div className="error center">
```

Comment:
What are the errors in this situation?

Reply:
Since there is nothing to interact with on the UI except clicking a button, how would the user fix these issues before proceeding? Can you demo this part?

Reply:
chatted in via video. resolved.